services/meta/client.go

```diff
@@ -71,7 +71,12 @@ func (c *Client) Close() error {
 	c.mu.Lock()
 	defer c.mu.Unlock()
 
-	close(c.closing)
+	select {
+	case <-c.closing:
+		return nil
+	default:
```

Comment:
This is in the `Close` method. I did this to ensure that if close gets called multiple times it doesn't try to close a channel that has already been closed.